### PR TITLE
Add support for in-memory persistence using Ballerina tables

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Related issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/4251

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [x] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/tests/sql-department-tests.bal [hunk 1]
```diff
@@ -1,6 +1,6 @@
-// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
 //
-// WSO2 Inc. licenses this file to you under the Apache License,
+// WSO3 Inc. licenses this file to you under the Apache License,
```
Comment (daneshk) [inline ballerina/tests/sql-department-tests.bal:3]:
```suggestion
// WSO2 Inc. licenses this file to you under the Apache License,
```

File: ballerina/tests/sql_rainier_generated_client.bal [hunk 2]
```diff
@@ -7,14 +23,14 @@ const BUILDING = "buildings";
...
-    *AbstractPersistClient;
+public client class SQLRainierClient {
+    *AbstractPersistSQLClient;
 
     private final mysql:Client dbClient;
 
     private final map<SQLClient> persistClients;
 
-    private final record {|Metadata...;|} metadata = {
+    private final record {|SQLMetadata...;|} metadata = {
```
Comment (daneshk) [inline ballerina/tests/sql_rainier_generated_client.bal:33]: We need this change in the generated client? @kalaiyarasiganeshalingam

File: ballerina/tests/sql_rainier_generated_client.bal [hunk 3]
```diff
@@ -7,14 +23,14 @@ const BUILDING = "buildings";
 const DEPARTMENT = "departments";
 const ORDER_ITEM = "orderitems";
 
-public client class RainierClient {
-    *AbstractPersistClient;
+public client class SQLRainierClient {
+    *AbstractPersistSQLClient;
```
Comment (daneshk) [inline ballerina/tests/sql_rainier_generated_client.bal:27]: We need this change in the generated client? @kalaiyarasiganeshalingam

File: ballerina/tests/sql-department-tests.bal (hunk 1 above)
Comment (kaneeldias) [reply ballerina/tests/sql-department-tests.bal]: Fixed in [945e4cb](https://github.com/ballerina-platform/module-ballerina-persist/pull/186/commits/945e4cbb91c854bcefc85fac4b7810c903a0125f)

File: ballerina/sql_client.bal [hunk 4]
```diff
@@ -444,5 +444,5 @@ public client class SQLClient {
 }
 
 # Represents the abstract persist client. This abstract object is used in the generated client.
-public type AbstractPersistClient distinct object {
+public type AbstractPersistSQLClient distinct object {
```
[the remark this answers is not in the document]
Comment (kaneeldias) [reply ballerina/sql_client.bal]: Removed and `AbstractPersistClient` has been moved to its own file in [3e3f198](https://github.com/ballerina-platform/module-ballerina-persist/pull/186/commits/3e3f198960dfca78bb811873709abe70d184a620)

File: ballerina/tests/sql_rainier_generated_client.bal (hunk 3 above)
Comment (kaneeldias) [reply ballerina/tests/sql_rainier_generated_client.bal]: Not needed

**Review by daneshk — DISMISSED** [review]

**Review by daneshk — APPROVED** [review]

**Pull request merged**